src/flows/context/schemaProvider.tsx

```diff
@@ -69,22 +127,40 @@ export const SchemaProvider: FC<Props> = React.memo(({children}) => {
   }, [data?.bucket?.id, lastBucket, update])
 
   useEffect(() => {
-    dispatch(startWatchDog())
-  }, [dispatch])
-
-  useEffect(() => {
-    if (!data.bucket) {
+    if (!data?.bucket) {
       return
     }
 
-    dispatch(getAndSetBucketSchema(data.bucket, flow.range))
-  }, [data?.bucket, dispatch, flow?.range])
-
-  const schema = useSelector(
-    (state: AppState) => state.flow.schema[data.bucket?.name]?.schema || {}
-  )
+    setLoading(RemoteDataState.Loading)
+
+    const range = formatTimeRangeArguments(flow?.range)
+
+    const text = `from(bucket: "${data.bucket.name}")
+|> range(${range})
+|> first()
+|> drop(columns: ["_value"])
+|> group()`
+
+    const result = query(text)
+    result
+      .then((response: FluxResult) => {
+        const schemaForBucket = parsedResultToSchema(response.parsed)
+        setSchema(schemaForBucket)
+        setLoading(RemoteDataState.Done)
+      })
+      .catch((error: RunQueryResult) => {
+        console.error('error: ', error)
+        setLoading(RemoteDataState.Error)
+        setSchema({})
+      })
+  }, [data?.bucket?.name, data?.bucket, flow?.range, query])
```

Comment:
just need data.bucket.name here right? should make the one on line 130 look for that as well

Reply:
since flow.range is an object, this might be called more than it needs to be because of the shallow compare and tendency to overwrite using partials in the contexts. formatTimeRangeArguments returns a string, which should be more stable / intentional than the object reference within the shallow compare, so pull range out of this useEffect and reference it from within